src/main/java/org/cristalise/kernel/persistency/outcome/Outcome.java

```diff
@@ -407,10 +407,10 @@ public String getFieldByXPath(String xpath) throws XPathExpressionException, Inv
      * Determines if the NodeList is actually a single field, an element with text data only
      *
      * @param elements NodeList
-     * @return if the NodeList has a single field or not
+     * @return if the NodeList has a single field of type ELEMENT_NODE or not
      */
-    public boolean hasSingleField(NodeList elements) {
-        return (elements != null && elements.getLength() == 1 && elements.item(0).getNodeType() == Node.ELEMENT_NODE);
+    private boolean hasSingleField(NodeList elements) {
+        return (elements != null && elements.getLength() > 0 && elements.item(0).getNodeType() == Node.ELEMENT_NODE);
```

Comment:
this change means that it would return true even for multiple elements. it's enough for the first one to be of type ELEMENT_NODE

Reply:
Enter issue to replace getElementsByTagname with a shallow search